libarchive/archive_read_support_format_mtree.c

```diff
@@ -1251,7 +1251,10 @@ parse_file(struct archive_read *a, struct archive_entry *entry,
 			mtree->fd = open(path, O_RDONLY | O_BINARY | O_CLOEXEC);
 			__archive_ensure_cloexec_flag(mtree->fd);
 			if (mtree->fd == -1 &&
-				(errno != ENOENT ||
+                            /* On Windows, attempting to open a file with an invalid name
```

Comment:
I assume the correct way to go would be using:
```c
#if defined(_WIN32) && !defined(__CYGWIN__)
```
instead of modifying code for all platforms.

Reply:
There you go, I've pushed an adjustment for this.